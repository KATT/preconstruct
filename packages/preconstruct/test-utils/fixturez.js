// @flow
// https://github.com/jamiebuilds/fixturez/
import path from "path";
import * as fsExtra from "fs-extra";
import onExit from "signal-exit";
import _fixturez from "fixturez";
import uuid from "uuid/v4";

function fixturez(cwd: string) {
  let { find } = _fixturez(cwd);
  let created = [];
  function temp(): string {
    let tempDir = path.join(
      __dirname,
      "node_modules",
      "fixture-copies",
      uuid()
    );
    fsExtra.ensureDirSync(tempDir);
    created.push(tempDir);
    return tempDir;
  }

  function copy(name: string): string {
    let dest = path.join(temp(), name);
    fsExtra.copySync(find(name), dest);
    return dest;
  }

  function cleanup() {
    let err;
    created.forEach(tempDir => {
      try {
        fsExtra.removeSync(tempDir);
      } catch (e) {
        err = e;
      }
    });
    created.length = 0;
    if (err) throw err;
  }

  onExit(cleanup);

  return {
    find,
    copy
  };
}

module.exports = fixturez;
