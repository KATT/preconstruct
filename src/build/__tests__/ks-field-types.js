// @flow
import build from "../";
import fixturez from "fixturez";
import { snapshotDirectory } from "../../../test-utils";

const f = fixturez(__dirname);

jest.mock("../../prompt");

test("field types", async () => {
  let tmpPath = f.copy("field-types");

  await build(tmpPath);

  await snapshotDirectory(tmpPath);
});
