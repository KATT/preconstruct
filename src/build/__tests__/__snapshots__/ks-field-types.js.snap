// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`field types: dist/chunk-c4d933aa.cjs.dev.js 1`] = `
"'use strict';

var Field = () => {
  return \\"some cool react component stuff\\";
};

exports.default = Field;
"
`;

exports[`field types: dist/chunk-c4d933aa.cjs.prod.js 1`] = `
"\\"use strict\\";
var Field = () => \\"some cool react component stuff\\";
exports.default = Field;
"
`;

exports[`field types: dist/field-types.cjs.dev.js 1`] = `
"'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true
});
let Text = { ...otherFieldTypeStuff,
  views: {
    Field: require.resolve('./chunk-c4d933aa.cjs.dev.js')
  }
};
exports.Text = Text;
"
`;

exports[`field types: dist/field-types.cjs.js 1`] = `
"'use strict';

if (process.env.NODE_ENV === \\"production\\") {
  module.exports = require(\\"./field-types.cjs.prod.js\\");
} else {
  module.exports = require(\\"./field-types.cjs.dev.js\\");
}
"
`;

exports[`field types: dist/field-types.cjs.prod.js 1`] = `
"\\"use strict\\";
Object.defineProperty(exports, \\"__esModule\\", { value: !0 });
let Text = {
  ...otherFieldTypeStuff,
  views: { Field: require.resolve(\\"./chunk-c4d933aa.cjs.prod.js\\") }
};
exports.Text = Text;
"
`;

exports[`field types: src/index.js 1`] = `
"export let Text = {
  ...otherFieldTypeStuff,
  views: {
    Field: importView(\\"./views/Field\\")
  }
};
"
`;

exports[`field types: src/views/Field.js 1`] = `
"export default () => {
  return \\"some cool react component stuff\\";
};
"
`;
